### DISCUSSION TITLE
Alert Based on keywords match from message field

### DISCUSSION THREAD

**Discussion by ramippala** [jertel/elastalert2#1157] [Q&A] [answered]
Hi, 

I was trying to create a rule, that'll trigger an alert when a word matches or the message field contains the specific keywords ("ERROR", "keyError"). so I created the rule. I tested it using "elastlalert-test-rule", but the result was 0 matches.

rule yml

```
name: Error Alert
type: blacklist
index: logs-*
compare_key: "message"
blacklist:
  - "keyError"
  - "ERROR"
ignore_null: true

filter:
  - query:
      query_string:
        query: "message:keyError OR message:ERROR"

alert:
- "email"
alert_text_type: alert_text_only
alert_text_args: ["host.hostname", "message", "@timestamp"]
alert_text: "Error in {0} API at {2} and Content is {1}."

email:
  - ""
  - ""

realert:
  minutes: 120

alert_subject_args:
  - host.hostname
  - "@timestamp"

alert_subject: Error in {0} API logs at {1}.
```

test-rule output:
![image](https://user-images.githubusercontent.com/25664553/231087140-8ff5df7a-f863-47db-ad3f-0476642ba0e5.png)

help would be appreciated

**Comment by jertel** [comment, marked as answer]
Blacklist rules are exact matches to the field value. You are more likely trying to do a partial match. Try removing the `compare_key:`, `ignore_null:`, and `blacklist:` parameters, and change the `type:` parameter to `any`.

**Reply by ramippala** [reply]
Cool, it's working as expected, Great thanks
